app/models/katello/content.rb

```diff
@@ -37,17 +37,20 @@ def propagate_name_change?
 
     def self.import_all
       Organization.all.each do |org|
+        cp_products = ::Katello::Resources::Candlepin::Product.all(org.label, [:id, :productContent])
+        prod_content_importer = Katello::ProductContentImporter.new
+
         org.products.each do |product|
-          begin
-            product_json = Katello::Resources::Candlepin::Product.get(org.label,
-                                                                  product.cp_id,
-                                                                  %w(productContent)).first
-            product_content_attrs = product_json['productContent']
-            Katello::Glue::Candlepin::Product.import_product_content(product, product_content_attrs)
-          rescue RestClient::NotFound
+          product_json = cp_products.find { |prod| prod['id'] == product.cp_id }
```

Comment:
I am ok with this change. However I am wondering if you want to do a hash with id as key instead.
Its an overkill if there are only like 10 products per org, but if there are 100s hash might be better,

Reply:
I think it's probably negligible since it's sandwiched in between an HTTP request and multiple database operations

Reply:
Hmm I saw this in one of the production databases
cp_products.count
=> 113
> o.products.count
=> 55

I think its worth making it a hash

Reply:
updated!